src/client.rs

```diff
@@ -430,10 +430,10 @@ impl<T: Read + Write> Client<T> {
     ///     };
     /// }
     /// ```
-    pub fn authenticate<A: Authenticator, S: AsRef<str>>(
+    pub fn authenticate(
         mut self,
-        auth_type: S,
-        authenticator: &A,
+        auth_type: impl AsRef<str>,
+        authenticator: &impl Authenticator,
```

Comment:
Ah, so, `Authenticator` I _think_ we may want to stay as `<A>`, since it's feasible that someone may want to specify the authenticator using turbofish syntax (which doesn't work with `:impl`). I don't know of a concrete case, I just have a feeling.